src/lib/server/websearch/markdown/types.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable-next-line no-shadow */
+export enum MarkdownElementType {
+	Header = "HEADER",
+	Paragraph = "PARAGRAPH",
+	BlockQuote = "BLOCKQUOTE",
+	CodeBlock = "CODE_BLOCK",
+
+	UnorderedList = "UNORDERED_LIST",
+	OrderedList = "ORDERED_LIST",
+	UnorderedListItem = "UNORDERED_LIST_ITEM",
+	OrderedListItem = "ORDERED_LIST_ITEM",
+}
+
+interface BaseMarkdownElement<T = MarkdownElementType> {
+	type: T;
+	content: string;
+	parent: HeaderElement | null;
```

Comment:
```suggestion
	parent?: HeaderElement;
```

we can make `parent` optional unless having an actual `NULL` value is preferred

Reply:
yeah I preferred `null` here because I felt it was more explicit in the lack of a parent than `undefined` which may indicate to a reader that the parent is optional